packages/route/src/Infrastructure/Doctrine/EventListener/RouteChangedUpdater.php

```diff
@@ -98,6 +98,7 @@ public function postFlush(PostFlushEventArgs $args): void
                 ->andWhere(\is_string($site) ? 'parent.site = :site' : 'parent.site IS NULL')
                 ->andWhere('parent.locale = :locale')
                 ->andWhere('(parent.slug = :newSlug OR parent.slug LIKE :oldSlugSlash)') // direct child is using newSlug already updated as we are in PostFlush, grand child use oldSlugWithSlash as not yet updated
+                ->andWhere('(child.slug LIKE :oldSlugSlash)') // ignore disconnected child routes in case of full tree edit
```

Comment:
this is the whole magic now we ignore the once which do not start the same way and are not a child